tensor2tensor/data_generators/generator_utils.py

```diff
@@ -338,13 +338,19 @@ def generate():
 
         # Use Tokenizer to count the word occurrences.
         with tf.gfile.GFile(filepath, mode="r") as source_file:
-          file_byte_budget = 3.5e5 if filepath.endswith("en") else 7e5
+          file_byte_budget = 1e6 if filepath.endswith("en") else 1e6
+          counter = 0
+          countermax = int(source_file.size() / 1e6)
```

Comment:
I understand the goal: each `countermax`th line is sampled, but
* I would suggest to make `countermax` lower, otherwise in 50% of cases we reach the end of file without reaching file_byte_budget bytes of sampled data (I expect line lengths are distributed randomly in the training data).
* Don't repeat the constant 1e6, use `file_byte_budget` instead.
* Add a comment about the intended goal: to get a representative sample from (possibly) unshuffled training data